doc/subport/sonic-sub-port-intf-hld.md

```diff
@@ -96,63 +102,98 @@ A sub port interface shall support the following features:
 * VRF
 * RIF counters
 * QoS setting inherited from parent physical port or port channel
-* mtu inherited from parent physical port or port channel
+* MTU: 
+  MTU inherited from parent physical port or port channel. 
+  If subinterface MTU is configured, MTU on subinterface will be configured with:
+  - If Subinterface MTU <= parent port MTU, configured subinterface MTU will be applied.
+  - If Subinterface MTU > parent port MTU, parent port MTU will be applied.
 * Per sub port interface admin status config
+  - Kernel subinterface netdev admin UP can be performed only if parent interface netdev is admin UP.
+    Hence subinterface admin UP is performed only after parent interface is admin UP.
 
 # 2 Schema design
 
 We introduce a new table "VLAN_SUB_INTERFACE" in the CONFIG_DB to host the attributes of a sub port interface.
 For APPL_DB and STATE_DB, we do not introduce new tables for sub port interfaces, but reuse existing tables to host sub port interface keys.
 
 ## 2.1 Configuration
-### 2.1.1 config_db.json
+### 2.1.1 Naming Convention for sub-interfaces:
+
+Since Kernel has netdevice name length restriction to 15, port channel sub-interfaces(Physical interfaces as well in case interface number > 99) cannot follow the same nomenclature as physical sub-interfaces. Hence Long name to  short name conversion needs to be performed for the subinterfaces.
```

Comment:
what's 99 here?

Reply:
If parent physical interface has interface number which exceeds 2 digits, subinterface name length will exceed 15 characters with existing subinterface naming convention. This results in failure of subinterface netdev creation in kernel. 

To be more specific, this subinterface netdev name length issue is not specific to PortChannel subinterfaces, but also applicable to physical subinterfaces.